specification/0041-hashed-mappings-specification.md

```diff
@@ -0,0 +1,143 @@
+# Hashed Mapping Specification
+
+## Summary
+
+This document intends to revise and formalize the process for creating a
+hashed mapping as originally proposed in the hashed mojmap [RFC 19](https://github.com/QuiltMC/rfcs/blob/master/rfc/0019-hashed-mojmap.md).
+
+## Motivation
+
+Hashed Mojmap is intended to replace intermediary as our runtime mappings.
+It provides some benefits over intermediary, most notably the reduced maintenance cost.
+
+One of the design goals of hashed was, that it is not built incrementally,
+but can always be creating from an existing official mapping set without knowledge of earlier hashed versions.
+This also means that it should be possible for anyone to generate hashed mappings,
+even if they don't plan to use the mappings-hasher designed by Quilt.
+In order to allow this, a specification for the hashed mapping is needed.
+
+In order to specify our intended behaviour, this document describes the process to generate a hashed mapping.
+Anything not specified in this RFC is considered a bug in the generation of hashed mappings.
+
+## Explanation
+
+Given an original mappings set and the corresponding classes, a new mapping set is created with the following properties:
+- All new mappings are a hash of the corresponding original mapping.
+- All new mappings are unique in the new mapping set where possible.
+- The new mappings must ensure that the resulting remapped jar can be verified and executed.
+- The new mapping set should be as resilient to changes in the original mapping set and corresponding classes as possible.
+
+In order to achieve this, the following operations are performed on the original mapping set:
+- Convert all class/method/field names into their corresponding "raw" name.
+- Ensure methods that are required to have the same name do so.
```

Comment:
In my opinion, better wording here be: `Ensure that methods within a name set have the same raw name`, to lead the reader to the section on name sets if they are curious as to what "methods that are required to have the same name" are applicable to.

Reply:
Is this good enough?
Im not sure if the name sets are a requirement for hashed or if they're an imterpretation of the requirements.

Reply:
> - Ensure methods that are required to have the same name in the jvm have the same "raw" name. (See "Method Name Sets".)

I don't think "that are required to have the same name in the JVM" is fully accurate or at the very least informational here, as it does not then explain _why_ the JVM requires these methods to have the same name.

I think it would be more accurate to say "that are in the same method override hierarchy" or something to that effect? As the method name sets seem to be the way of saying "in a tree of methods which override each other over multiple classes, all of their raw names will be the same as the root of that tree -- that is, the method which does not override any method and from which all other methods in the tree override."

Also, it may be beneficial to qualify any reference to `name` in the spec as either an "original name", "raw name", or "hashed name", for ease of identifying what "name" is being referenced at any point.

Reply:
"override hierarchy" seems reasonable. I just don't want to include the jvm spec in this RFC.

And good idea to qualify name. I'll update the PR.